tools/data-api/internal/repositories/services.go

```diff
@@ -1,227 +1,411 @@
-						},
-						Resource:        "VirtualMachine",
-						ResourceIdName:  "VirtualMachineId",
-						ResourceName:    "VirtualMachine",
-						SchemaModelName: "VirtualMachineSchema",
-						SchemaModels: map[string]TerraformSchemaModelDefinition{
-							"VirtualMachineSchema": {
-								Fields: map[string]TerraformSchemaFieldDefinition{
-									"VirtualMachineId": {
-										ObjectDefinition: TerraformSchemaFieldObjectDefinition{
-											NestedObject:  nil,
-											ReferenceName: nil,
-											Type:          StringTerraformSchemaFieldType,
-										},
-										Computed: false,
-										ForceNew: true,
-										HclName:  "virtual_machine_id",
-										Optional: false,
-										Required: true,
-										Documentation: TerraformSchemaDocumentationDefinition{
-											Markdown: "blah",
-										},
-										Validation: nil,
-									},
-									"Name": {
-										ObjectDefinition: TerraformSchemaFieldObjectDefinition{
-											NestedObject:  nil,
-											ReferenceName: nil,
-											Type:          StringTerraformSchemaFieldType,
-										},
-										Computed: false,
-										ForceNew: true,
-										HclName:  "name",
-										Optional: false,
-										Required: true,
-										Documentation: TerraformSchemaDocumentationDefinition{
-											Markdown: "blah",
-										},
-										Validation: nil,
-									},
-								},
-							},
-						},
-						Tests: TerraformResourceTestsDefinition{
-							BasicConfiguration:          "basictest",
-							RequiresImportConfiguration: "importtest",
-							CompleteConfiguration:       nil,
-							Generate:                    false,
-							OtherTests:                  nil,
-							TemplateConfiguration:       nil,
-						},
-						UpdateMethod: pointer.To(MethodDefinition{
-							Generate:         true,
-							MethodName:       "VirtualMachineCreateOrUpdate",
-							TimeoutInMinutes: 30,
-						}),
-					},
-				},
-			},
-
-			Generate: true,
-		},
+func (s *ServicesRepositoryImpl) ProcessServiceDefinitions(serviceName string) (*ServiceDetails, error) {
+	versions, err := listSubDirectories(fmt.Sprintf("%s/%s", s.directory, serviceName))
+	if err != nil {
+		return nil, fmt.Errorf("retrieving versions: %+v", err)
+	}
+
+	versionDefinitions := make(map[string]*ServiceApiVersionDetails, 0)
+
+	if versions != nil {
+		for _, version := range *versions {
+			resources, err := listSubDirectories(fmt.Sprintf("%s/%s/%s", s.directory, serviceName, version))
+			if err != nil {
+				return nil, fmt.Errorf("retrieving resources for %s: %+v", version, err)
+			}
+			versionDetails, err := s.ProcessVersionDefinitions(serviceName, version, *resources)
+			if err != nil {
+				return nil, fmt.Errorf("processing version definitions for %s: %+v", version, err)
+			}
+			versionDefinitions[version] = versionDetails
+		}
+	}
+
+	return &ServiceDetails{
+		Name:             serviceName,
+		ResourceProvider: fmt.Sprintf("Microsoft.%s", serviceName),
```

Comment:
Resource Providers don't have to be within the `Microsoft.` namespace - we've already got some which aren't (`Nginx.Nginx` etc) - so this'd need to be stored alongside the data

In addition, the Service name that we're using can differ (e.g. `VMware` [pandora] vs `Avs` [azure-rest-api-specs]) so that it's clearer as to what this is - therefore these two are intentionally different